src/base/Roles/AccountantWithRateProviders.sol

```diff
@@ -366,22 +474,93 @@ contract AccountantWithRateProviders is Auth, IRateProvider {
+            uint256 protocolFees =
+                totalDeposits.mulDivDown(lendingInfo.protocolFeeRate * timeElapsed, SECONDS_PER_YEAR * BASIS_POINTS);
+            totalFees += protocolFees;
+        }
     }
 
     // ========================================= INTERNAL HELPER FUNCTIONS =========================================
+    /**
+     * @notice Checkpoint protocol fees only
+     * @dev Updates feesOwedInBase with accrued protocol fees
+     */
+    function _checkpointProtocolFees() internal {
+        if (vault.totalSupply() > 0 && lendingInfo.protocolFeeRate > 0) {
+            uint256 timeElapsed = block.timestamp - lendingInfo.lastAccrualTime;
+            if (timeElapsed > 0) {
+                uint256 totalDeposits = vault.totalSupply().mulDivDown(accountantState.exchangeRate, ONE_SHARE);
+                uint256 protocolFees =
+                    totalDeposits.mulDivDown(lendingInfo.protocolFeeRate * timeElapsed, SECONDS_PER_YEAR * BASIS_POINTS);
+                accountantState.feesOwedInBase += uint128(protocolFees);
+            }
+        }
+    }
+
+    /**
+     * @notice Checkpoint both interest and protocol fees
+     * @dev Updates exchange rate with interest and feesOwedInBase with protocol fees
+     */
+    function _checkpointInterestAndFees() internal {
+        if (vault.totalSupply() > 0 && lendingInfo.lendingRate > 0) {
+            uint256 timeElapsed = block.timestamp - lendingInfo.lastAccrualTime;
+            if (timeElapsed > 0) {
+                uint256 totalDeposits = vault.totalSupply().mulDivDown(accountantState.exchangeRate, ONE_SHARE);
+
+                // Calculate and apply interest to exchange rate
+                uint256 interestAccrued =
+                    totalDeposits.mulDivDown(lendingInfo.lendingRate * timeElapsed, SECONDS_PER_YEAR * BASIS_POINTS);
+
+                if (interestAccrued > 0) {
+                    uint256 rateIncrease = interestAccrued.mulDivDown(ONE_SHARE, vault.totalSupply());
+                    accountantState.exchangeRate += uint96(rateIncrease);
+                }
+
+                // Calculate and store protocol fees
+                uint256 protocolFees =
+                    totalDeposits.mulDivDown(lendingInfo.protocolFeeRate * timeElapsed, SECONDS_PER_YEAR * BASIS_POINTS);
+                accountantState.feesOwedInBase += uint128(protocolFees);
+            }
+        }
+    }
+
     /**
      * @notice Used to change the decimals of precision used for an amount.
      */
-    function changeDecimals(uint256 amount, uint8 fromDecimals, uint8 toDecimals) internal pure returns (uint256) {
-        if (fromDecimals == toDecimals) {
-            return amount;
-        } else if (fromDecimals < toDecimals) {
-            return amount * 10 ** (toDecimals - fromDecimals);
+    function changeDecimals(uint256 _amount, uint8 _fromDecimals, uint8 _toDecimals) internal pure returns (uint256) {
```

Comment:
Internal function should start with _ like https://github.com/Ozean-L2/nucleus-boring-vault/pull/2/files#diff-b40fbd70955504478372516aed7abfb854b66064a9ced21b8861558e192d0883R531